helm_drydock/drivers/node/maasdriver/driver.py

```diff
@@ -0,0 +1,306 @@
+
+    def execute_task(self):
+        task_action = self.task.action
+
+        self.orchestrator.task_field_update(self.task.get_id(),
+                            status=hd_fields.TaskStatus.Running,
+                            result=hd_fields.ActionResult.Incomplete)
+
+        self.maas_client = MaasRequestFactory(self.driver_config['api_url'],
+                                              self.driver_config['api_key'])
+
+        site_design = self.orchestrator.get_effective_site(self.task.design_id,
+                                                           self.task.site_name)
+
+        if task_action == hd_fields.OrchestratorAction.CreateNetworkTemplate:
+            # Try to true up MaaS definitions of fabrics/vlans/subnets
+            # with the networks defined in Drydock
+            design_networks = site_design.networks
+
+            subnets = maas_subnet.Subnets(self.maas_client)
+            subnets.refresh()
+
+            result_detail = {
+                'detail': []
+            }
+
+            for n in design_networks:
+                exists = subnets.query({'cidr': n.cidr})
+
+                subnet = None
+
+                if len(exists) > 0:
+                    subnet = exists[0]
+
+                    subnet.name = n.name
+                    subnet.dns_servers = n.dns_servers
+
+                    vlan_list = maas_vlan.Vlans(self.maas_client, fabric_id=subnet.fabric)
+                    vlan_list.refresh()
+
+                    vlan = vlan_list.select(subnet.vlan)
+
+                    if vlan is not None:
+                        if ((n.vlan_id is None and vlan.vid != 0) or 
+                           (n.vlan_id is not None and vlan.vid != n.vlan_id)):
+
+                            # if the VLAN name matches, assume this is the correct resource
+                            # and it needs to be updated
+                            if vlan.name == n.name:
+                                vlan.set_vid(n.vlan_id)
+                                vlan.mtu = n.mtu
+                                vlan.update()
+                            else:
+                                vlan_id = n.vlan_id if n.vlan_id is not None else 0
+                                target_vlan = vlan_list.query({'vid': vlan_id})
+                                if len(target_vlan) > 0:
+                                    subnet.vlan = target_vlan[0].resource_id
+                                else:
+                                    # This is a flag that after creating a fabric and
+                                    # VLAN below, update the subnet
+                                    subnet.vlan = None
+                    else:
+                        subnet.vlan = None
+            
+                    # Check if the routes have a default route
+                    subnet.gateway_ip = n.get_default_gateway()
+
+
+                    result_detail['detail'].append("Subnet %s found for network %s, updated attributes"
```

Comment:
More verbose logging like this, all over, everywhere, for everything.

I'd recommend the `logging` module and to make liberal use of log.debug(...) statements and log.info, log.warn where appropriate.  This can shuffle the output anywhere that is appropriate.  I can provide examples if needed.

Capturing important detail like this and shuttling it back through results is great, but the ability to run the service in --debug or debug=True (yet another config.conf item!) and see virtually every decision is really really helpful for you and for others.